client/ayon_core/lib/execute.py

```diff
@@ -209,6 +209,14 @@ def run_ayon_launcher_process(*args, **kwargs):
         # - fill more if you find more
         env = clean_envs_for_ayon_process(os.environ)
 
+    if add_sys_paths:
+        new_pythonpath = list(sys.path)
+        for path in env.get("PYTHONPATH", "").split(os.pathsep):
+            if not path or path in new_pythonpath:
+                continue
```

Comment:
Both sys.path and `PYTHONPATH` may be large lists, so we're now doing a `O1` lookup in `new_pythonpath` which may take some time due to that.

It may be better if this could be an ordered set or use a separate mechanism on the side to track whether a path was already included. That may yield performance benefits.

Reply:
It's at about 0.002 seconds. Do you really think it should be optimized more?

Reply:
Let's not optimize before it's used :)

Reply:
> It's at about 0.002 seconds. Do you really think it should be optimized more?

Depends on the amount of entries on your PYTHONPATH I'd say? It'd be exponential and we must admit have MANY on our PYTHONPATH. And I dare say that anyone using `rez` will cry because there ever single dependency is its own entry in pythonpath.

At 1000 entries it's at 0.006 seconds.
At 10.000 entries it's at 0.6 seconds.
At 100.000 entries it's at 64 seconds.

Whereas just by making it a `set` as a lookup takes the 100.000 entries down to 0.01 seconds.

```python
num = 10000
values = list(range(num))
import time

s = time.time()
for x in values:
    if x in values:
        pass
e = time.time()
print(e-s)

values_lookup = set(values)
import time

s = time.time()
for x in values:
    if x in values_lookup:
        pass
e = time.time()
print(e-s)
```

Reply:
Haha - man, you should check out some of the `rez` issues. There were literally cases where they were going beyond the limits of what OS'es allowed for these paths and hence stuff getting dismissed accidentally. So they swapped over the technique to specialized Python files you can drop on Pythonpath and alike pointing to many paths. Which interestingly enough also optimized startup times by insane amounts (for whatever reason).

Anyway, will mark resolved.